### ISSUE TITLE
Unknown transactions on remote and local nodes

### ISSUE THREAD

**Issue by som-dev** [theQRL/QRL#1324]
I am running v0.63.0 and I started noticing some addresses had mismatching quanta when comparing my local block explorer vs. the explorer.theqrl.org block explorer.  My local node appeared to be in-sync with the remote in that they both agreed on the most recent block number and time.

I wrote a script to retrieve an address_state and then attempt to retrieve each transaction from the address_state to see if I could account for the differences.  I found two issues:
1) some addresses had more transactions on the remote node than my local node
2) on both remote and local nodes, some transactions could not be retrieved via the API or the block explorer

I have more if needed but below is one concrete example for address Q010600a69e0eb760fbeb9ca69508f66b0b305ff6ad452214bcde8ffcebf122cf9e4cb7d405b242.
```
~/dev/som-dev/qrl-audit$ ./qrl-audit.py --host 127.0.0.1 address --address 010600a69e0eb760fbeb9ca69508f66b0b305ff6ad452214bcde8ffcebf122cf9e4cb7d405b242
INFO:root:Connected to 127.0.0.1:9009
INFO:root:balance: 3501076745880
INFO:root:found 4840 transactions
ERROR:root:txhash 1966a6039294d6cc297927796b4e6b4e589c56de6dadc12c1435d30155c3e55b for addr 010600a69e0eb760fbeb9ca69508f66b0b305ff6ad452214bcde8ffcebf122cf9e4cb7d405b242 does not exist
ERROR:root:txhash 9b135db8a8b09087a9c82a6bf2cdedaea7ea344d186a5b4052453f334104cd51 for addr 010600a69e0eb760fbeb9ca69508f66b0b305ff6ad452214bcde8ffcebf122cf9e4cb7d405b242 does not exist

~/dev/som-dev/qrl-audit$ ./qrl-audit.py --host 35.177.60.137 address --address 010600a69e0eb760fbeb9ca69508f66b0b305ff6ad452214bcde8ffcebf122cf9e4cb7d405b242
INFO:root:Connected to 35.177.60.137:9009
INFO:root:balance: 3514363280088
INFO:root:found 4842 transactions
ERROR:root:txhash eff278e7450e66fcea31b652cfa5e2e1021dfc2a3d7da1727e0103849aa81639 for addr 010600a69e0eb760fbeb9ca69508f66b0b305ff6ad452214bcde8ffcebf122cf9e4cb7d405b242 does not exist
ERROR:root:txhash 1966a6039294d6cc297927796b4e6b4e589c56de6dadc12c1435d30155c3e55b for addr 010600a69e0eb760fbeb9ca69508f66b0b305ff6ad452214bcde8ffcebf122cf9e4cb7d405b242 does not exist
ERROR:root:txhash 9b135db8a8b09087a9c82a6bf2cdedaea7ea344d186a5b4052453f334104cd51 for addr 010600a69e0eb760fbeb9ca69508f66b0b305ff6ad452214bcde8ffcebf122cf9e4cb7d405b242 does not exist
ERROR:root:txhash 02231c57c40007a5de9f40b3b8ae0c3027f59a5387e5b1f961649f5a3f26a028 for addr 010600a69e0eb760fbeb9ca69508f66b0b305ff6ad452214bcde8ffcebf122cf9e4cb7d405b242 does not exist
```

**Comment by cyyber** [comment]
Can you rerun the same command and also let us know the exact blockheight of your node?

**Comment by som-dev** [comment]
My node's last block as reported in the log at 12:04 CT:
```
2018-05-16 12:04:30,291|0.63.0|synced| INFO : Added Block #26918 195c83277345a12456831e73960d81e1b7cfc521752b5c73bd5e4b0f44240100
```
Here is a rerun of my script making the api calls at 12:04 CT (the output looks the same as the first run):
```
~/dev/som-dev/qrl-audit$ ./qrl-audit.py --host 127.0.0.1 address --address 010600a69e0eb760fbeb9ca69508f66b0b305ff6ad452214bcde8ffcebf122cf9e4cb7d405b242
INFO:root:Connected to 127.0.0.1:9009
INFO:root:balance: 3501076745880
INFO:root:found 4840 transactions
ERROR:root:txhash 1966a6039294d6cc297927796b4e6b4e589c56de6dadc12c1435d30155c3e55b for addr 010600a69e0eb760fbeb9ca69508f66b0b305ff6ad452214bcde8ffcebf122cf9e4cb7d405b242 does not exist
ERROR:root:txhash 9b135db8a8b09087a9c82a6bf2cdedaea7ea344d186a5b4052453f334104cd51 for addr 010600a69e0eb760fbeb9ca69508f66b0b305ff6ad452214bcde8ffcebf122cf9e4cb7d405b242 does not exist

~/dev/som-dev/qrl-audit$ ./qrl-audit.py --host 35.177.60.137 address --address 010600a69e0eb760fbeb9ca69508f66b0b305ff6ad452214bcde8ffcebf122cf9e4cb7d405b242
INFO:root:Connected to 35.177.60.137:9009
INFO:root:balance: 3514363280088
INFO:root:found 4842 transactions
ERROR:root:txhash eff278e7450e66fcea31b652cfa5e2e1021dfc2a3d7da1727e0103849aa81639 for addr 010600a69e0eb760fbeb9ca69508f66b0b305ff6ad452214bcde8ffcebf122cf9e4cb7d405b242 does not exist
ERROR:root:txhash 1966a6039294d6cc297927796b4e6b4e589c56de6dadc12c1435d30155c3e55b for addr 010600a69e0eb760fbeb9ca69508f66b0b305ff6ad452214bcde8ffcebf122cf9e4cb7d405b242 does not exist
ERROR:root:txhash 9b135db8a8b09087a9c82a6bf2cdedaea7ea344d186a5b4052453f334104cd51 for addr 010600a69e0eb760fbeb9ca69508f66b0b305ff6ad452214bcde8ffcebf122cf9e4cb7d405b242 does not exist
ERROR:root:txhash 02231c57c40007a5de9f40b3b8ae0c3027f59a5387e5b1f961649f5a3f26a028 for addr 010600a69e0eb760fbeb9ca69508f66b0b305ff6ad452214bcde8ffcebf122cf9e4cb7d405b242 does not exist
```

**Comment by som-dev** [comment]
That script connecting to the remote takes awhile, now my node reports:
```
2018-05-16 12:16:19,377|0.63.0|synced| INFO : >>>Received block from 35.177.60.137:9000 26935 0c1a9d7f1f054497be9cc6c1f70d304f9d61bf3303d97f5e3e6623675d7a0000
2018-05-16 12:16:19,667|0.63.0|synced| INFO : Added Block #26935 0c1a9d7f1f054497be9cc6c1f70d304f9d61bf3303d97f5e3e6623675d7a0000
2018-05-16 12:17:18,862|0.63.0|synced| INFO : >>>Received block from 35.177.60.137:9000 26936 e97e6520d675ba8fd9bcd365b73fa23f073d000aa8a312a570cedd72bce40000
2018-05-16 12:17:19,137|0.63.0|synced| INFO : Added Block #26936 e97e6520d675ba8fd9bcd365b73fa23f073d000aa8a312a570cedd72bce40000
```

**Comment by cyyber** [comment]
@som-dev Can you run following command on your local node
qrl state
Please paste the output of the above command here. This will let us know your current chain state.

**Comment by som-dev** [comment]
~/dev/som-dev/qrl-audit$ date; qrl state
Wed May 16 13:31:07 CDT 2018
info {
  version: "0.63.0"
  state: SYNCED
  num_connections: 3
  num_known_peers: 116
  uptime: 170267
  block_height: 27024
  block_last_hash: "\2255\260|\353\007\210\214\254\033\2040\231fG\251\346\221\331-\217\311p\255\315\303\216\231R\330\000\000"
  network_id: "Outside Context Problem"
}

**Comment by cyyber** [comment]
I see the block explorer have the correct value. As I started another node that synced from block number 0 and that node shows the same value. I will need the qrl.log file from you, so that I can further look into what happened with your localnode.

**Comment by som-dev** [comment]
I uploaded all qrl logs to https://transfer.sh/t6Sny/qrl_logs_issue1324.zip

**Comment by som-dev** [comment]
I noticed my local node was receiving data from 104.237.3.185 this morning.
When I ran my custom tool to pull every transaction in the address_state it reported no issues.  So as you mention, there are definitely good nodes out there.
```
~/dev/som-dev/qrl-audit$ ./qrl-audit.py --host 104.237.3.185 address --address 010600a69e0eb760fbeb9ca69508f66b0b305ff6ad452214bcde8ffcebf122cf9e4cb7d405b242
INFO:root:Connected to 104.237.3.185:9009
INFO:root:balance: 3534174940222
INFO:root:found 4845 transactions
```

**Comment by cyyber** [comment]
Is this possible for you to share your leveldb data?

**Comment by som-dev** [comment]
Sure, no problem!  What is the best way to do that?  Stop the node and zip up the entire ~/.qrl/data folder?

**Comment by cyyber** [comment]
yes stop the node and zip up entire ~/.qrl/data folder
also share your node config

**Comment by som-dev** [comment]
After multiple failed attempts at uploading a 600MB+ file to various file sharing services, I finally recalled the "split" command which brings me back...
I have 6 parts:
https://transfer.sh/X4zs1/qrl_data_issue1234.zip.partaa
https://transfer.sh/H2d1l/qrl_data_issue1234.zip.partab
https://transfer.sh/hL4s4/qrl_data_issue1234.zip.partac
https://transfer.sh/FTjgg/qrl_data_issue1234.zip.partad
https://transfer.sh/gfGYK/qrl_data_issue1234.zip.partae
https://transfer.sh/X8UUG/qrl_data_issue1234.zip.partaf

Here's the md5sum to know you've got them all reconstructed properly:
```
$ md5sum qrl_data_issue1324.zip
86382cda1a80d92f48add9a3854f1296  qrl_data_issue1324.zip
$ cat qrl_data_issue1324.zip.part* > reconstructed.zip
$ md5sum reconstructed.zip 
86382cda1a80d92f48add9a3854f1296  reconstructed.zip
```
Here is my config:
```
$ more ~/.qrl/config.yml 
mining_enabled : False
mining_thread_count : 1
```

**Comment by cyyber** [comment]
Thanks for reporting. Issue has been fixed in PR #1334 .
Fix will be included in the next release of QRL node.
